src/engine/plugins/table.cpp

```diff
@@ -67,7 +67,14 @@ Status TablePlugin::HandleRequest(const datafacade::ContiguousInternalMemoryData
         return Error("TooBig", "Too many table coordinates", result);
     }
 
-    auto snapped_phantoms = SnapPhantomNodes(GetPhantomNodes(facade, params));
+    auto phantom_nodes = GetPhantomNodes(facade, params);
+
+    if (phantom_nodes.size() != params.coordinates.size())
```

Comment:
You can use the size of `params.coordinate` to sprint a more helpful error message (e.g. "Could not find a matching segment for the i-th coordinate").